lib/internal/console/global.js

```diff
@@ -44,8 +44,13 @@ for (const prop of ReflectOwnKeys(Console.prototype)) {
   ReflectDefineProperty(globalConsole, prop, desc);
 }
 
+const {
+  getColorDepth
+} = require('internal/tty');
 globalConsole[kBindStreamsLazy](process);
-globalConsole[kBindProperties](true, 'auto');
+globalConsole[kBindProperties](true, () => {
+  return getColorDepth() > 2 || 'auto';
```

Comment:
I think this would also need to check that the streams actually refer to TTYs? Do you remember why this change was necessary specifically (just since I’m surprised, it seems unrelated to the rest of the changes).

In any case, this definitely seems like a better solution than https://github.com/nodejs/node/pull/32308.

Reply:
Yes, https://github.com/nodejs/node/blob/f1940824d9f0e769d4ecce07307d4b126cd8dbb4/test/sequential/test-util-debug.js has FORCE_COLORS in a non-tty child.

Reply:
Okay, then I misread this – `'auto'` should be the right value to pass for colors here. The `kGetInspectOptions` method should do the right thing by checking the stream’s `getColorDepth()` property.

Reply:
Yeah, so … I tried this locally, with only the changes from `lib/internal/util/debuglog.js`, and I think it’s actually fine to just adjust the test here, rather than the `console` code itself. The test already previously monkey-patched `process.stderr.hasColors` to make it look as if `process.stderr` was a TTY, you can just add another monkey-patch to add `process.stderr.isTTY` as well and be done with that.

Reply:
ok